src/main/java/gross/citibike/service/CitiBikeFunctions.java

```diff
@@ -1,15 +1,19 @@
 package gross.citibike.service;
 
+import software.amazon.awssdk.services.s3.S3Client;
+
 import java.util.List;
 
 public class CitiBikeFunctions {
-
+    private StationsCache stationsCache;
+    private final String bucketName = "gross.citibike";
     private List<StationResponse.StationInfo> stations;
     private List<StatusResponse.Status> statuses;
 
     public CitiBikeFunctions() {
```

Comment:
This should be a member variable in your RequestHandler.
You are recreating the StationsCache each time handleRequest() is called.